### PR TITLE
Query

### PR DESCRIPTION
**Pull request by Sparsh-Goyal-2000**

### CODE REVIEW

File: app/models/cart.rb [hunk 1]
```diff
@@ -1,5 +1,7 @@
 class Cart < ApplicationRecord
     has_many :line_items, dependent: :destroy
+    has_many :products, through: :line_items
+    has_many :enabled_products, -> {where enabled: true}, class_name: 'Products', through: :line_items
```
Comment (Sparsh-Goyal-2000) [inline app/models/cart.rb:4]: Change for Query Interface 1st:
Make a scope for all the enabled products

File: app/models/order.rb [hunk 2]
```diff
@@ -10,6 +11,8 @@ class Order < ApplicationRecord
     }
     validates :name, :address, :email, presence: true
     validates :pay_type, inclusion: pay_types.keys    
+
+    scope :by_date, -> (from = Time.zone.now.midnight, to = Time.zone.now.end_of_day) { where created_at: from..to }
```
Comment (Sparsh-Goyal-2000) [inline app/models/order.rb:15]: Change for Query Interface 2nd:
Add another scope in order which accept two dates and return orders placed between
those dates
○ Order.by_date(from, to)
○ Order.by_date : without arguments, should return order for the current day
○ user.orders.by_date should work and it should return orders placed by user
between given dates.

File: app/models/product.rb [hunk 3]
```diff
@@ -1,44 +1,46 @@
...
 
-  has_many :line_items
+  has_many :line_items, dependent: :restrict_with_error
   has_many :orders, through: :line_items
+  has_many :carts, through: :line_items
 
   before_destroy :ensure_not_referenced_by_any_line_item
   after_initialize :set_defaults
 
+  scope :enabled, -> { where enabled: true }
```
Comment (Sparsh-Goyal-2000) [inline app/models/product.rb:42]: Change for Query Interface 1st:
Make a scope for all the enabled products

**Comment by apekshit-tewathia** [comment]
why does this PR have so many file changes?
Fix it so I see changes related to query interface only

**Pull request closed without merging**